QgisModelBaker/gui/panel/session_panel.py

```diff
@@ -136,6 +136,14 @@ def __init__(
 
         self.is_skipped_or_done = False
 
+    def hideEvent(self, event):
```

Comment:
@m-kuhn Maybe you have an input here. We have a QWidget in a QWizardPage in a QWizard and this in a QDialog. When the dialog is closed we notice this only when this object here is destroyed (and then it's too late to do anything) or when it gets hidden. If it's hidden for any other reason (like clicking "back" in wizard") it makes sense as well that the process is terminated.

But is this save?

Reply:
Would it not be safer to handle this logic at QDialog closeEvent level? Calling there cancel_session on the relevant instance?
But i did not look very deeply so sorry if not mega relevant. But more generally speaking i prefer to take actions at the source of the event than relying on signal/slot connections.

Reply:
I thought about that, but we do not receive a closeEvent on the QWidget and not on the QWizardPage and the QWizard. Only the QDialog. There the approach would be to override it and then go to the wizard / wizard pages / widgets to kill the process. Would be possible for sure, but not sure if it's better.

Reply:
Hm... Maybe I need to follow another approach anyway. Since on terminating the process the function returns to the dialog that is no longer existent. I think the way to go is blocking the closing until everything is cleaned up.